client-tags/icon.md

```diff
@@ -0,0 +1,85 @@
+---
+
+## Notes for implementing work-in-progress version
+
+This is a work-in-progress specification.
+
+Software implementing this work-in-progress specification MUST NOT use the
+unprefixed `+icons` tag name. Instead, implementations SHOULD use the
+`+draft/icons` tag name to be interoperable with other software
+implementing a compatible work-in-progress version.
+
+The final version of the specification will use an unprefixed tag name.
+
+## Introduction
+
+This specification defines a client-only message tag to indicate message icons
+
+## Motivation
+
+To allow for richer display of messages with added visual context, a method for client and bot users to attach custom icons to messages is needed. This method is intended for message specific icons. For persistent avatars, a metadata key would be more appropriate.
+
+Visual clients are given a way to choose an appropriate icon resource for the display context, with a method for providing multiple icon sources, with different dimensions.
+
+A content type hint is also provided, to allow clients to filter unsupported image formats without downloading the resource.
+
+## Architecture
+
+### Dependencies
+
+Clients wishing to use this tag MUST negotiate the [`draft/message-tags`](../core/message-tags-3.3.html) capability with the server.
+
+### Format
+
+The icons tag is sent by a client with the client-only prefix `+` and its value is a JSON encoded array of image objects that reference icon resources for the message.
+
+    +draft/icons=<json>
+
+Image objects MAY contain any of the following members:
+
+| Member           | Description |
+| :--------------- | :---------- |
+| `src` (string)   | The Uniform Resource Indicator (URI) of an icon resource |
+| `sizes` (string) | A space separate list of pixel image dimensions in `wxh` format supported by the resource. Clients MAY use this to select an appropriate icon resource for the context and pixel density of the display |
+| `type` (string)  | A media type hint for the icon resource. Clients MAY use this to filter out unsupported icon types without downloading the resource |
+
+When an icon `src` is an HTTP resource, it SHOULD be referenced with an HTTPS protocol where available.
+
+When the icon is encoded as an [RFC2397 Data URI](https://tools.ietf.org/html/rfc2397) the length of the URI SHOULD be no more than 1024.
+
+Icon dimensions SHOULD be square.
+
+## Client implementation considerations
+
+This section is non-normative
+
+Client tags are untrusted data and icon URIs might point to any arbitrary resource. Some examples of unexpected icon conditions and potential mitigations:
+
+* Large file sizes: Validate file sizes and abort if appropriate limits are reached.
+* Unexpected dimensions: Scale or crop images to fit within allocated user interface space.
+* Unknown formats: Validate content types and reject unaccepted image formats.
+* Malicious content: Allow users to block specific icon URIs or icons from specific users.
```

Comment:
Can add a security consideration about that, but worth noting that clients that embed image links have the same issue.